presto-accumulo/src/main/java/com/facebook/presto/accumulo/io/AccumuloRecordCursor.java

```diff
@@ -0,0 +1,537 @@
+            case StandardTypes.INTEGER:
+                return serializer.getInt(fieldToColumnName[field]);
+            case StandardTypes.TIME:
+                return serializer.getTime(fieldToColumnName[field]).getTime();
+            case StandardTypes.TIMESTAMP:
+                return serializer.getTimestamp(fieldToColumnName[field]).getTime();
+            default:
+                throw new PrestoException(INTERNAL_ERROR,
+                        "Unsupported type " + getType(field));
+        }
+    }
+
+    @Override
+    public Object getObject(int field)
+    {
+        Type type = getType(field);
+        checkArgument(Types.isArrayType(type) || Types.isMapType(type),
+                "Expected field %s to be a type of array or map but is %s", field, type);
+
+        if (Types.isArrayType(type)) {
+            return serializer.getArray(fieldToColumnName[field], type);
+        }
+
+        return serializer.getMap(fieldToColumnName[field], type);
+    }
+
+    @Override
+    public Slice getSlice(int field)
+    {
+        Type type = getType(field);
+
+        if (type instanceof VarbinaryType) {
+            return Slices.wrappedBuffer(serializer.getVarbinary(fieldToColumnName[field]));
+        }
+        else if (type instanceof VarcharType) {
+            return Slices.utf8Slice(serializer.getVarchar(fieldToColumnName[field]));
+        }
+        else {
+            throw new PrestoException(INTERNAL_ERROR,
+                    "Unsupported type " + type);
+        }
+    }
+
+    @Override
+    public void close()
+    {
+        scan.close();
+        nanoEnd = System.nanoTime();
+    }
+
+    /**
+     * Gets a Boolean value indicating whether or not the scanner should only return row IDs
+     * <p>
+     * This can occur in cases such as SELECT COUNT(*) or the table only has one column.
+     * Presto doesn't need the entire contents of the row to count them,
+     * so we can configure Accumulo to only give us the first key/value pair in the row
+     *
+     * @param rowIdName Row ID column name
+     * @return True if scanner should retriev eonly row IDs, false otherwise
+     */
+    private boolean retrieveOnlyRowIds(String rowIdName)
+    {
+        return cHandles.size() == 0 || (cHandles.size() == 1 && cHandles.get(0).getName().equals(rowIdName));
+    }
+
+    /**
+     * Checks that the given field is one of the provided types
+     *
+     * @param field Ordinal of the field
+     * @param expected An array of expected types
+     * @throws IllegalArgumentException If the given field does not match one of the types
+     */
+    private void checkFieldType(int field, Type... expected)
+    {
+        Type actual = getType(field);
+        for (Type t : expected) {
+            if (actual.equals(t)) {
+                return;
+            }
+        }
+
+        checkArgument(false, "Expected field %s to be a type of %s but is %s", field,
```

Comment:
`throw new IllegalArgumentException` explicitly instead of calling `checkArgument(false,...)` which is misleading.